src/wallet/wallet.cpp

```diff
@@ -1508,6 +1512,9 @@ CBlockIndex* CWallet::ScanForWalletTransactions(CBlockIndex* pindexStart, bool f
             } else {
                 ret = nullptr;
             }
+            if (pindex == pindexStop) {
```

Comment:
See the discussion about the stop argument: https://github.com/bitcoin/bitcoin/pull/7061#issuecomment-158808001